### PR TITLE
Virthost fixes

### PR DESCRIPTION
**Pull request by cbosdo**
A bunch of fixes for the new virt host module.
Ruled out the 12 family: too much of a pain with libguestfs.

### CODE REVIEW

**Comment by cbosdo** [comment]
Note: need to wait for testsuite fix before merging

File: salt/repos/repos.d/SLE-Module-Server-Applications-SLE-15-x86_64-Pool.repo [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[SLE-Module-Server-Applications-SLE-15-x86_64-Pool]
+name=SLE-Module-Server-Applications-SLE-15-x86_64-Pool
+type=rpm-md
+enabled=1
+baseurl=http://{{ grains.get("mirror") | default("download.suse.de/ibs", true) }}/SUSE/Products/SLE-Module-Server-Applications/15/x86_64/product/
```
Comment (moio) [inline salt/repos/repos.d/SLE-Module-Server-Applications-SLE-15-x86_64-Pool.repo]: Please also add this repo to `minima.yaml`

File: salt/virthost/init.sls [hunk 2]
```diff
@@ -142,6 +143,10 @@ disk-image-template.qcow2:
   file.managed:
     - name: /var/testsuite-data/disk-image-template.qcow2
     - source: {{ grains['hvm_disk_image'] }}
+    {% if grains['hvm_disk_image_hash'] %}
+    - source_hash: {{ grains['hvm_disk_image_hash'] }}
```
Comment (moio) [inline salt/virthost/init.sls]: Is it possible not to specify this parameter? I have vague memories that it was mandatory.

Comment (cbosdo) [reply salt/virthost/init.sls]: This is mandatory only for http(s) and ftp protocols.

File: salt/repos/repos.d/SLE-Module-Server-Applications-SLE-15-x86_64-Pool.repo (hunk 1 above)
Comment (cbosdo) [reply salt/repos/repos.d/SLE-Module-Server-Applications-SLE-15-x86_64-Pool.repo]: For some reason it is already in `minima.yaml`

**Pull request merged**